File: app/src/main/java/com/nathaniel/motus/cavevin/controller/CellarInputUtils.java
package com.nathaniel.motus.cavevin.controller;

import android.content.Context;
import android.widget.Toast;

import com.nathaniel.motus.cavevin.R;

import java.util.ArrayList;
import java.util.Arrays;

public class CellarInputUtils {
    //a utilitary class to check all inputs

//    **********************************************************************************************
//    Static declaration
//    **********************************************************************************************

    private static final ArrayList<String> forbiddenCharacters= new ArrayList<>(Arrays.asList("\"", "\\","{","}","[","]"));
    private static final String replacementCharacter="*";


//    **********************************************************************************************
//    Constructor, private not to instantiate the class
//    **********************************************************************************************

    private CellarInputUtils() {
    }

//    **********************************************************************************************
//    Input check subs
//    **********************************************************************************************

    public static String replaceForbiddenCharacters(Context context, String inputString){
        //replace forbidden characters by replacementCharacter

        String initialInputString=inputString;
        for (int i=0;i<forbiddenCharacters.size();i++){
            inputString=inputString.replace(forbiddenCharacters.get(i),replacementCharacter);
        }

        if (initialInputString.compareTo(inputString)!=0){
            Toast.makeText(context, R.string.cellar_input_utils_forbidden_characters,Toast.LENGTH_SHORT).show();
        }

        return inputString;
    }

}

File: app/src/main/java/com/nathaniel/motus/cavevin/controller/CellarInputUtils.kt
package com.nathaniel.motus.cavevin.controller

import android.content.Context
import com.nathaniel.motus.cavevin.controller.CellarInputUtils
import android.widget.Toast
import com.nathaniel.motus.cavevin.R
import java.util.*

object CellarInputUtils {
    //a utilitary class to check all inputs
    //    **********************************************************************************************
    //    Static declaration
    //    **********************************************************************************************
    private val forbiddenCharacters = ArrayList(Arrays.asList("\"", "\\", "{", "}", "[", "]"))
    private const val replacementCharacter = "*"

    //    **********************************************************************************************
    //    Input check subs
    //    **********************************************************************************************
    @JvmStatic
    fun replaceForbiddenCharacters(context: Context?, inputString: String): String {
        //replace forbidden characters by replacementCharacter
        var inputString = inputString
        val initialInputString = inputString
        for (i in forbiddenCharacters.indices) {
            inputString = inputString.replace(forbiddenCharacters[i], replacementCharacter)
        }
        if (initialInputString.compareTo(inputString) != 0) {
            Toast.makeText(
                context,
                R.string.cellar_input_utils_forbidden_characters,
                Toast.LENGTH_SHORT
            ).show()
        }
        return inputString
    }
}